File: benchmarks/fair/arrayiter.cs
using System;
// Dizi uzerinde deyimsel yineleme: doldur + topla. C#'ta uzunluk KAPTA
// (a.Length) — C'deki gibi elle tasinan `n` degil.
class arrayiter {
    static void Main() {
        long n = 5000000;
        string e = Environment.GetEnvironmentVariable("BENCH_N");
        if (e != null && long.TryParse(e, out long v) && v > 0) n = v;
        long[] a = new long[n];
        long t = 0;
        for (int i = 0; i < a.Length; i++) a[i] = i;
        for (int i = 0; i < a.Length; i++) t += a[i];
        Console.WriteLine(t);
    }
}

File: benchmarks/fair/arrayiter.cpp
// Dizi uzerinde deyimsel yineleme: doldur + topla. C++'ta uzunluk KAPTA
// (a.size()) — C'deki gibi elle tasinan `n` degil; kural bu: her dil kendi
// deyimsel uzunluk erisimini kullaniyor.
#include <cstdio>
#include <cstdlib>
#include <vector>
int main(){
    const char*e=std::getenv("BENCH_N"); long n=e?std::atol(e):0; if(n<=0)n=5000000;
    std::vector<long long> a(n); long long t=0;
    for(size_t i=0;i<a.size();i++) a[i]=(long long)i;
    for(size_t i=0;i<a.size();i++) t+=a[i];
    std::printf("%lld\n",t); return 0;
}
